snapcraft/meta/snap_yaml.py

```diff
@@ -175,11 +177,13 @@ def write(project: Project, prime_dir: Path, *, arch: str, arch_triplet: str):
         assumes.add("command-chain")
 
     environment = _populate_environment(project.environment, prime_dir, arch_triplet)
+    # mypy false positive: at this point project.version is a string
+    version = _process_version(project.version)  # type: ignore
```

Comment:
instead of this I would do something like this to indicate that the project at this point needs to have been processed

```suggestion
    if project.version is None:
        raise ValueError("'project.version' cannot be None")
    version = _process_version(project.version)  # type: ignore
```

Reply:
or handle the case inside _process_version

Reply:
Handling it inside `_process_version`.